plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -309,6 +309,8 @@ public PostgreSqlClient(
                 .add(new RewriteIn())
                 .withTypeClass("integer_type", ImmutableSet.of("tinyint", "smallint", "integer", "bigint"))
                 .withTypeClass("numeric_type", ImmutableSet.of("tinyint", "smallint", "integer", "bigint", "decimal", "real", "double"))
+                .withTypeClass("string_type", ImmutableSet.of("char", "varchar"))
+                .map("reverse(value: string_type)").to("REVERSE(value)")
```

Comment:
Would it work on columns which are mapped to varchar i.e money being mapped to varchar.

Reply:
It does not have sense to use it for money :)

Reply:
The serious answer is that we cannot control it. 
Also, in most cases it does not corrupt data - it has an immediate trino-runtime, remote-compile-time exception.
The best idea I came so far is to cover it with `join.pushdown.disabled-functions=reverse,cast` config/session property.
Also it's a wider issue - it does not relate to the `reverse(value)` expression itself, but to the external scope as well, when different type of return is expected. e.g. `reverse(value) = value`
https://github.com/trinodb/trino/pull/21939#discussion_r1598358639

Reply:
> Also, in most cases it does not corrupt data - it has an immediate trino-runtime, remote-compile-time exception.

Mostly yes - I was thinking on re-writing `RewriteVaraible` rule which rewrites the column which has a native mapping and the column which supports pushdown operation.

Reply:
> In some JDBC connector like PostgreSQL - we try to map unknown column type or special column type like money as varchar - so in this case we should somehow restrict the expression rewrite - I think it can be fixed by avoid rewriting the variables which are synthetically mapped.

It's tricky, as information about initial remote type is already lost, at the moment we try to rewrite for pushdown.

Reply:
> RewriteVaraible

Thank you for pointing out, will take a look.

Reply:
Yes, this will be happening in this case as in all other cases, take this predicate pushdown, for example:

```java
TestTable table = new TestTable(
                onRemoteDatabase(),
                "test_money_functions",
                "(col money, col2 varchar)",
                List.of("8, 'test'", "9, 'test'");

"SELECT col FROM " + table.getName() + " WHERE col LIKE '%$8%';
```

Connector attempts to push down LIKE on `money` and fails:

> Caused by: org.postgresql.util.PSQLException: ERROR: operator does not exist: money ~~ character varying
  Hint: No operator matches the given name and argument types. You might need to add explicit type casts.
  Position: 74